### PR TITLE
TRD renamed HitType to Hit, own file and moved to DataFormats

### PR DESCRIPTION
**Pull request by bazinski**
Move class HitType out of Detectors.h into its own header file in DataFormats.
This removes a dependency of base on simulation for the TrackletTransformer
It also neatens up the code.
Renamed class to simply o2::trd::Hit instead of o2::trd::HitType.

### CODE REVIEW

**Comment by martenole** [comment]
Did you forget to add the new HitType header file?

**Comment by bazinski** [comment]
@martenole doh, juggling too many branches. hopefully fine now

**Review by martenole — DISMISSED** [review]
Looks good

File: Detectors/TRD/simulation/src/Digitizer.cxx [hunk 1]
```diff
@@ -19,6 +19,8 @@
 #include "TRDBase/PadPlane.h"
 #include "TRDBase/PadResponse.h"
 
+#include "DataFormatsTRD/HitType.h"
```
Comment (jolopezl) [inline Detectors/TRD/simulation/src/Digitizer.cxx]: Just style, but please reorder the headers. I think DataFormats should precede TRDBase.

File: Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "DetectorsBase/BaseDPLDigitizer.h"
 #include "TRDBase/Calibrations.h"
 #include "DataFormatsTRD/TriggerRecord.h"
+#include "DataFormatsTRD/HitType.h"
```
Comment (jolopezl) [inline Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx]: Same as above, please check the order of the headers.

**Review by jolopezl — CHANGES_REQUESTED** [review]
I appreciate this change a lot; it was very needed and was for a long time in my backlog; thanks for doing the change. I had in mind for this change renaming the class from `HitType` to `Hit`. There is no need for the `Type` in `HitType`; also, it will read much better when you use `o2::trd::Hit` than `o2::trd::HitType` anywhere in the code.

File: Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx (hunk 2 above)
Comment (bazinski) [reply Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx]: Done.
Given that the style follows what was originally there with regards to TriggerRecord and there are TRDsimulation includes between TRDbase includes, root includes amongst O2 includes, which style guide are we following?

File: Detectors/TRD/simulation/src/Digitizer.cxx (hunk 1 above)
Comment (bazinski) [reply Detectors/TRD/simulation/src/Digitizer.cxx]: comment as below

File: Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx (hunk 2 above)
Comment (jolopezl) [reply Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx]: We should follow Google Style: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

> For example, the includes in google-awesome-project/src/foo/internal/fooserver.cc might look like this:
> ```
> #include "foo/server/fooserver.h"
> 
> #include <sys/types.h>
> #include <unistd.h>
> 
> #include <string>
> #include <vector>
> 
> #include "base/basictypes.h"
> #include "base/commandlineflags.h"
> #include "foo/server/bar.h"
> ```

> Exception:
> 
> Sometimes, system-specific code needs conditional includes. Such code can put conditional includes after other includes. Of course, keep your system-specific code small and localized. Example:
> ```
> #include "foo/public/fooserver.h"
> 
> #include "base/port.h"  // For LANG_CXX11.
> 
> #ifdef LANG_CXX11
> #include <initializer_list>
> #endif  // LANG_CXX11
> ```

Still, I think this is not compulsory in O2. Sometimes forcing the order of includes can cause dependency problems.

**Comment by bazinski** [comment]
@jolopezl HitType renamed to Hit, compiles fine. Only one script seems to be affected.

File: Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx (hunk 2 above)
Comment (bazinski) [reply Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx]: Ja of course order can matter

**Review by tdietel — DISMISSED** [review]
Good idea.

**Review by martenole — APPROVED** [review]

**Review by tdietel — APPROVED** [review]

**Pull request merged**